cpp/include/nvtext/replace.hpp

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <cudf/column/column.hpp>
+#include <cudf/scalar/scalar.hpp>
+#include <cudf/strings/strings_column_view.hpp>
+
+//! NVText APIs
+namespace nvtext {
+/**
+ * @addtogroup nvtext_replace
+ * @{
+ */
+
+/**
+ * @brief Replaces specified tokens with corresponding replacement strings.
+ *
+ * Tokens are identified in each string and if any match the specified `targets`
+ * strings, they are replaced with corresponding `replacements` string such that
+ * if `targets[i]` is found, then it is replaced by `replacements[i]`.
+ *
+ * The `delimiter` may be zero or more characters. If the `delimiter` is empty,
+ * whitespace (character code-point <= ' ') is used for identifying tokens.
+ * Also, any consecutive delimiters found in a string are ignored.
+ *
+ * @code{.pseudo}
+ * Example:
+ * s = ["this is me", "theme music"]
+ * tgt = ["is", "me"]
+ * rpl = ["+", "_"]
+ * result = replace_tokens(s,tgt,rpl)
+ * result is now ["this + _", "theme music"]
+ * @endcode
+ *
+ * A null input element at row `i` produces a corresponding null entry
+ * for row `i` in the output column.
+ *
+ * There is no guaranteed order for replacing the targets. This means if
```

Comment:
When would this be relevant?

Reply:
Yes, you are right. This will not happen here. I will remove the comment.